### DISCUSSION TITLE
How to disable pre-pending of the Wiki Space key to migrated pages

### DISCUSSION THREAD

**Discussion by timeross** [WikiTransformationProject/wikitraccs-releases#79] [Q&A]
We always do a 1 space to 1 Sharepoint site migration, so I don't want the space key to be prepended to the Titles of all the migrated pages.  I'm not finding anywhere in the settings or config files to disable this.  Has anyone discovered how to do this?

**Comment by heinrich-ulbricht** [comment]
Currently this is not possible and would require an additional setting to be implemented.

**Reply by timeross** [reply]
Thanks for the quick reply, Heinrich.  

I'm pretty comfortable tweaking files a bit.  Isn't the function to grab the space key and build the Sharepoint page name buried somewhere in one of the many files in the Console folder?  Couldn't I just go in and tweak the function so that instead of grabbing the space key, it just grabbed "" (blank)?

This is a licensed version of your software.  It's an impressive tool you built!  Thanks for all your work.

**Reply by heinrich-ulbricht** [reply]
Thanks for the feedback!  This is much appreciated.

The file name logic is hardcoded, no chance in editing it. This would be a good feature, though.

The only thing I am wary about is that page links between migrated pages in SharePoint are generated based on this naming scheme. So changing the scheme would have to be done before migrating any pages at all; or complicate things.

What could work is renaming pages with PnP PowerShell afterwards. I think SharePoint takes care of adjusting page links. But not 100% sure. Would have to test this.

**Reply by timeross** [reply]
Okay.  I haven't had anyone complain about it yet, so maybe it isn't that important to my users.  I can play around with PowerShell if anyone does request it.

I definitely think that this would be a good setting to add to a future version (option to enable/disable pre-pending).

Thanks for the info!

**Reply by L3m0nRoCk** [reply]
reply on this:
I Tested renaming the pages, by removing the Space kay and pageID from the migrated pages. Sharepoint does take care of internal links between the pages. so renaming after the migration is complete is completely fine. (if you have a 404 not found, make sure you republished the page and refresh the browser)

I havent tested renaming programatically via a sharepoint api like Microsoft Graph. But I dont see why that wont work.
I will test this soon, and try to remember to edit this reply with the answere :)

**Reply by heinrich-ulbricht** [reply]
Thanks for chiming in!

**Reply by heinrich-ulbricht** [reply]
Note: this blog post explains the naming scheme: [Migrated SharePoint Page Names Explained](https://www.wikitransformationproject.com/blog/2024/11/16/migrated-sharepoint-page-names-explained/).
